portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,13 +38,40 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    List<String> commentMessages = new ArrayList<String>();
-    commentMessages.add("This website looks real stylish!");
-    commentMessages.add("I also am a lover of chicken alfredo!");
-    commentMessages.add("Wow! How long did this take you?");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timeOfComment", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Comment> listOfComments = new ArrayList<Comment>();
+    // Creates a Comment object for each entity that was previously ever posted. 
+    for (Entity commentEntity : results.asIterable()) {
```

Comment:
Is the type just Entity? That looks wrong. You must've setup an actual specific entity for the comments, no?

Reply:
I think so, the comments are just stored together under the same kind, "Comment".